app/security.py

```diff
@@ -113,10 +113,10 @@ async def context_processor(request):
     }
 
 
-async def check_permission(request):
+async def check_permission(request) -> Session:
```

Comment:
They are two separate concerns which do what they say on the tin:

- check_permission is concerned with making sure the user has entered a valid UAC or raising an unauthorised error of some kind. Hence it is included in security.py.  It is clear it is for use with the remember and forget functions. It could have been implemented in many ways. We simply use an Identity stored in their Redis session, their session expires they will also no longer be authenticated. It is a performance optimisation to return their session to remove the need to go to Redis to load the session again.   
- get_existing_session returns the existing user session, whether or not they have entered a valid UAC previously. The function's sole concern is to ensure that only an existing session is returned or session timeout is raised. I would call this after check_permission but that would not be optimal for performance reasons.
Combining the functions would not reflect two separate areas of concern are being dealt with and add slight complication. It would not be as clear, having to pass a parameter perhaps to show whether you expect a user to have entered a UAC or not?

Reply:
yeah I understand.  My main concern here was that a method named 'check_x' does not suggest that you are going to return something.  `check_permission` now has an important function (returning the session) that is not advertised by its name.  I understand why you have done it that way, but am never keen on functions / methods that do extras that wouldn't be clear from their name.

Reply:
Renamed to `get_permitted_session` as requested.